### PR TITLE
planning: introduce and implement dual warm up by osqp for open space

### PR DESCRIPTION
**Pull request by deidaraho**

### CODE REVIEW

File: modules/planning/open_space/trajectory_smoother/dual_variable_warm_start_problem.cc [hunk 1]
```diff
@@ -41,94 +41,117 @@ bool DualVariableWarmStartProblem::Solve(
...
-  app->Options()->SetStringValue("print_timing_statistics",
-      planner_open_space_config_.dual_variable_warm_start_config().
-      ipopt_config().ipopt_print_timing_statistics());
-  app->Options()->SetStringValue("alpha_for_y",
-      planner_open_space_config_.dual_variable_warm_start_config().
-      ipopt_config().ipopt_alpha_for_y());
-  app->Options()->SetStringValue("recalc_y", planner_open_space_config_.
-      dual_variable_warm_start_config().ipopt_config().ipopt_recalc_y());
-  // for qp problem speed up
   if (planner_open_space_config_.dual_variable_warm_start_config().
```
Comment (Capri2014) [inline modules/planning/open_space/trajectory_smoother/dual_variable_warm_start_problem.cc]: use an enum to chose between "OSQP", "IPOPT", and "IPOPTQP" please.

File: modules/planning/open_space/trajectory_smoother/dual_variable_warm_start_problem.cc [hunk 2]
```diff
@@ -41,94 +41,117 @@ bool DualVariableWarmStartProblem::Solve(
...
-      qp_format()) {
-    app->Options()->SetStringValue("mehrotra_algorithm", "yes");
-  }
-
-  Ipopt::ApplicationReturnStatus status = app->Initialize();
-  if (status != Ipopt::Solve_Succeeded) {
-    AERROR
-        << "*** Dual variable wart start problem error during initialization!";
+      enable_osqp()) {
```
Comment (Capri2014) [inline modules/planning/open_space/trajectory_smoother/dual_variable_warm_start_problem.cc]: use an enum to chose between "OSQP", "IPOPT", and "IPOPTQP" please.

Comment (deidaraho) [reply modules/planning/open_space/trajectory_smoother/dual_variable_warm_start_problem.cc]: > use an enum to chose between "OSQP", "IPOPT", and "IPOPTQP" please.

It is done.

File: modules/planning/open_space/trajectory_smoother/BUILD [hunk 3]
```diff
@@ -61,6 +62,29 @@ cc_library(
...
+    deps = [
+        "//modules/common/configs:vehicle_config_helper",
+        "//modules/common/math",
+        "//modules/common/util",
+        "//modules/planning/common:planning_gflags",
+        "//modules/planning/proto:planner_open_space_config_proto",
+        "//modules/planning/proto:planning_proto",
+        "@osqp",
+        "@eigen",
+        "@ipopt",
```
Comment (Capri2014) [inline modules/planning/open_space/trajectory_smoother/BUILD]: remove ipopt dependency for this one.

File: modules/planning/open_space/trajectory_smoother/dual_variable_warm_start_problem.cc (hunk 2 above)
Comment (Capri2014) [reply modules/planning/open_space/trajectory_smoother/dual_variable_warm_start_problem.cc]: use ENUM, not number

File: modules/planning/open_space/trajectory_smoother/BUILD (hunk 3 above)
Comment (deidaraho) [reply modules/planning/open_space/trajectory_smoother/BUILD]: > remove ipopt dependency for this one.

done

**Comment by Capri2014** [comment]
Test needed for new OSQP interface

**Comment by deidaraho** [comment]
> Test needed for new OSQP interface

I will write one soon.

**Comment by deidaraho** [comment]
> Test needed for new OSQP interface

done

**Pull request merged**